### PR TITLE
Mepts 350 changes in monthly report misau b7

### PR DESCRIPTION
**Pull request by cfaife**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java [hunk 1]
```diff
@@ -979,12 +979,13 @@ public CohortDefinition getNumberOfPatientsWhoAbandonedArtDuringPreviousMonthFor
     SqlCohortDefinition cd = new SqlCohortDefinition();
     cd.setName("Number of patients who Abandoned the ART during the current month");
     cd.addParameter(new Parameter("location", "Location", Location.class));
-    cd.addParameter(new Parameter("onDate", "On Date", Date.class));
+    cd.addParameter(new Parameter("onODate", "End Date", Date.class));
```
Comment (amosh101) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: Hi, there could be a typo on the parameter onDate.

Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: yes @amosh101  fixing it right now, thanks

**Review by amosh101 — CHANGES_REQUESTED** [review]
Just a small change on B7A onDate variable and the PR is good.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java (hunk 1 above)
Comment (ynurmahomed) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java]: There is a typo. I suggest using `onOrBefore` as the name (also update query).

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java [hunk 2]
```diff
@@ -402,29 +402,83 @@ public static String getPatientsWithFirstClinicalConsultationOnTheSameDateAsPreA
...
+            + "                             WHERE  pa.voided = 0   "
+            + "                                 AND enc.voided = 0   "
+            + "                                 AND obs.voided = 0   "
+            + "                                 AND obs.concept_id =  ${returnVisitDateForArvDrugConcept}   "
+            + "                                 AND obs.value_datetime IS NOT NULL   "
+            + "                                 AND enc.encounter_type = ${arvPharmaciaEncounterType}   "
+            + "                                 AND enc.location_id = :location   "
+            + "                                 AND enc.encounter_datetime <= :onDate   "
+            + "                             GROUP  BY pa.patient_id) fila   "
+            + "                         INNER JOIN encounter e on  "
```
Comment (ynurmahomed) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java:447]: This join does not seem necessary, what is it for?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java [hunk 3]
```diff
@@ -402,29 +402,83 @@ public static String getPatientsWithFirstClinicalConsultationOnTheSameDateAsPreA
...
+            + "                             e.patient_id = fila.patient_id and  "
+            + "                             e.encounter_datetime = fila.encounter_datetime AND  "
+            + "                             e.encounter_type = ${arvPharmaciaEncounterType} AND  "
+            + "                             e.location_id = :location AND  "
+            + "                             e.voided = 0  "
+            + "                         INNER JOIN obs o on  "
```
Comment (ynurmahomed) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java:453]: This join does not seem necessary.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java [hunk 4]
```diff
@@ -402,29 +402,83 @@ public static String getPatientsWithFirstClinicalConsultationOnTheSameDateAsPreA
...
-            + " ) final WHERE DATE_ADD(final.encounter_date, INTERVAL 90 DAY) < :onDate ";
-    return String.format(
-        query, pharmacyEncounterType, mastercardEncounterType, drugPickupConceptId);
+        "SELECT final.patient_id   "
+            + "             from(   "
+            + "                 SELECT   "
+            + "                     most_recent.patient_id, Date_add(Max(most_recent.value_datetime), interval 60 day) final_encounter_date   "
+            + "                 FROM   (SELECT fila.patient_id, o.value_datetime from (  "
+            + "                             SELECT enc.patient_id,   "
+            + "                                 Max(enc.encounter_datetime)  encounter_datetime  "
```
Comment (ynurmahomed) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java:432]: Shouldn't it be `Max(enc.encounter_datetime)  **value_datetime**`?

**Review by ynurmahomed — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java (hunk 2 above)
Comment (cfaife) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java:447]: this is needed to get  other attribuites of  encounter's  table and join with maximum encounters

**Review by ynurmahomed — APPROVED** [review]

**Pull request merged**